[new] Aiven Client Install instructions for MST (#1502)
timescale/docs: mst/aiven-client.md/aiven-client-install.md

---
title: Aiven Client for Managed Service for TimescaleDB
excerpt: Install and configure Aiven Command Line tool for fully managed services on AWS, Azure, or GCP.
product: mst
---

# Aiven Client for fully managed services

Aiven Client is a command line tool for fully managed services. To use Aiven
Client you need to first create an authentication token and then configure it to
connect to your Managed Service for TimescaleDB using the command line.

## Create an authentication token in Managed Service for TimescaleDB

To connect to Managed Service for TimecaleDB using Aiven Client you need to
create an authentication token.

<procedure>

> 

Reviewer: ```suggestion

### Creating an authentication token in Managed Service for TimescaleDB

```

> 1.  Login to your [Managed Service for TimescaleDB portal][mst-login].

Reviewer: ```suggestion
1.  Log in to your [Managed Service for TimescaleDB portal][mst-login].
```

> 1.  Click `User Information` icon in the top left corner.

Reviewer: ```suggestion
1.  Click `User Information` in the top right corner.
```

> 1.  In the `User Profile` page, select the `Authentication`tab.

Reviewer: ```suggestion
1.  In the `User Profile` page, navigate to the `Authentication`tab.
```

> 1.  Click the `Generate Token` button.

Reviewer: ```suggestion
1.  Click `Generate Token`.
```

1.  In the `Generate access token` dialog, type a descriptive name for the
    token and leave the rest of the fields blank.
1.  Copy the generated authentication token and save it.

</procedure>

> ## Install and configure Aiven Client and connect to Managed Service for TimescaleDB

Reviewer: ```suggestion
## Install the Aiven Client
```

> The `avn` utility is a Python package, so you can install using pip on a Linux,
> MacOS or a Windows system which have Python installed, using `pip install
> aiven-client` command. For more information about installaing Aiven Client, see
> the [getting started][aiven-github] documentation.

Reviewer: ```suggestion
Aiven Client is provided as a Python package, so you can install it on a Linux,
MacOS, or Windows system using `pip`, if you have already installed Python. 

Use this command: 

```bash
pip install aiven-client
```

For more information about installing the Aiven Client, see the 
[Aiven][aiven-github] documentation.
```

The GH webUI has broken my suggestion, but I'm sure you get the idea.

> ### Configuring Aiven Client to connect to Managed Service for TimescaleDB

Reviewer: ```suggestion
## Configure Aiven Client to connect to Managed Service for TimescaleDB
```

> To access Managed Service for TimescaleDB you need an authentication token.
> Aiven Client uses this authentication token to access your services on Managed
> Service for TimescaleDB.

Reviewer: ```suggestion
To access Managed Service for TimescaleDB with the Aiven Client, you need an 
authentication token. Aiven Client uses this authentication token to access your 
services on Managed Service for TimescaleDB.
```

<procedure>

> 

Reviewer: ```suggestion

### Configuring Aiven Client to connect to Managed Service for TimescaleDB

```

1.  Change to the install directory that contains the configuration files:

    ```bash
    cd ~/.config/aiven/
    ```

> 1.  Open the `aiven-credentials.json` using any editor and update the following

Reviewer: ```suggestion
1.  Open the `aiven-credentials.json` using any editor and update these
```

>     lines with your Managed Service for TimescaleDB `User email` and
>     `authentication token` that you generated :

Reviewer: ```suggestion
    lines with your Managed Service for TimescaleDB `User email`, and the
    `authentication token` that you generated :
```

    ```bash
    {
    "auth_token": "ABC1+123...TOKEN==",
    "user_email": "your.email@timescale.com"
    }
    ```

1.  Save the `aiven-credentials.json` file.

1.  To verify that you can access your services on Managed Service for
    TimescaleDB, type:

    ```bash
    avn project list
    ```

>     A list of all your projects appear.

Reviewer: ```suggestion
    This command shows a list of all your projects:
```

    ```bash
    
    PROJECT_NAME       DEFAULT_CLOUD            CREDIT_CARD
    =================  =======================  ===========

>     internal-90d0      timescale-aws-us-east-1  N/A
>     timescale-6d6d     timescale-aws-us-east-1  N/A
>     docs-test-project  timescale-aws-us-east-1  N/A

Reviewer: ```suggestion
    project-xxxx      timescale-aws-us-east-1  N/A
    project-yyyy     timescale-aws-us-east-1  N/A
    project-zzzz.    timescale-aws-us-east-1  N/A
```

Reviewer: Or, alternatively, we just remove the example output for this command. I'm not sure I'm happy having it show "N/A" for the credit card field.

Author: should we make that credit card 'xxxx-xxxx-xxxx-xxxx'?

Reviewer: Yeah, that might work 👍

Author: done!!!!

# Aiven Client for fully managed services

Aiven Client is a command line tool for fully managed services. To use Aiven
Client you need to first create an authentication token and then configure it to
connect to your Managed Service for TimescaleDB using the command line.

## Create an authentication token in Managed Service for TimescaleDB

To connect to Managed Service for TimecaleDB using Aiven Client you need to
create an authentication token.

<procedure>

### Creating an authentication token in Managed Service for TimescaleDB

1.  Log in to your [Managed Service for TimescaleDB portal][mst-login].
1.  Click `User Information` in the top right corner.
1.  In the `User Profile` page, navigate to the `Authentication`tab.
1.  Click `Generate Token`.
1.  In the `Generate access token` dialog, type a descriptive name for the
    token and leave the rest of the fields blank.
1.  Copy the generated authentication token and save it.

</procedure>

## Install the Aiven Client

Aiven Client is provided as a Python package, so you can install it on a Linux,
MacOS, or Windows system using `pip`, if you have already installed Python. 

Use this command: 

```bash
pip install aiven-client

> 

Reviewer: Need to add in the rest here ...

Reviewer: Wasn't there another line after the code block?

Author: Not it is a just one line command

Reviewer: Nope, found it:

```suggestion
For more information about installing the Aiven Client, see the
[Aiven][aiven-github] documentation.
```

Use this command: 

```bash
pip install aiven-client
```

## Configure Aiven Client to connect to Managed Service for TimescaleDB

To access Managed Service for TimescaleDB with the Aiven Client, you need an 
authentication token. Aiven Client uses this authentication token to access your 
services on Managed Service for TimescaleDB.

<procedure>

### Configuring Aiven Client to connect to Managed Service for TimescaleDB

1.  Change to the install directory that contains the configuration files:

    ```bash
    cd ~/.config/aiven/
    ```

1.  Open the `aiven-credentials.json` using any editor and update these
    lines with your Managed Service for TimescaleDB `User email`, and the
    `authentication token` that you generated :

    ```bash
    {
    "auth_token": "ABC1+123...TOKEN==",
    "user_email": "your.email@timescale.com"
    }
    ```

1.  Save the `aiven-credentials.json` file.

1.  To verify that you can access your services on Managed Service for
    TimescaleDB, type:

    ```bash
    avn project list
    ```

    This command shows a list of all your projects:

    ```bash
    
    PROJECT_NAME       DEFAULT_CLOUD            CREDIT_CARD

>     =================  =======================  ===========

Reviewer: Might need to adjust these to line up as well.

Author: adjusted
